slang-api/src/main/java/org/sonarsource/slang/impl/IntegerLiteralTreeImpl.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.slang.impl;
+
+import org.sonarsource.slang.api.IntegerLiteralTree;
+import org.sonarsource.slang.api.TreeMetaData;
+
+public class IntegerLiteralTreeImpl extends LiteralTreeImpl implements IntegerLiteralTree {
+
+  public IntegerLiteralTreeImpl(TreeMetaData metaData, String value) {
+    super(metaData, value);
+  }
+
+  @Override
+  public boolean isOctal() {
```

Comment:
I think that interpretation of the value is ultimately language specific and should be responsibility of AST converter specific for the language. SLang tree should be created by explicitly saying what kind of integer literal is it.